core/src/main/java/com/linecorp/armeria/common/HttpRequestBuilder.java

```diff
@@ -0,0 +1,397 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.linecorp.armeria.common.HttpHeaderNames.CONTENT_LENGTH;
+import static com.linecorp.armeria.common.HttpHeaderNames.COOKIE;
+import static java.util.Objects.requireNonNull;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+import javax.annotation.Nullable;
+
+import com.google.errorprone.annotations.FormatMethod;
+import com.google.errorprone.annotations.FormatString;
+
+import com.linecorp.armeria.common.FixedHttpRequest.EmptyFixedHttpRequest;
+import com.linecorp.armeria.common.FixedHttpRequest.OneElementFixedHttpRequest;
+import com.linecorp.armeria.common.FixedHttpRequest.TwoElementFixedHttpRequest;
+
+import io.netty.util.AsciiString;
+
+/**
+ * Builds a new {@link HttpRequest}.
+ */
+final class HttpRequestBuilder {
+
+    // TODO(tumile): Add content(Publisher).
+
+    private final RequestHeadersBuilder requestHeadersBuilder = RequestHeaders.builder();
+    private final HttpHeadersBuilder httpTrailersBuilder = HttpHeaders.builder();
+    private final QueryParamsBuilder queryParamsBuilder = QueryParams.builder();
+    private final Map<String, Object> pathParams = new HashMap<>();
+    private final List<Cookie> cookies = new ArrayList<>();
+    @Nullable
+    private HttpData content;
+    @Nullable
+    private String path;
+
+    /**
+     * Shortcut to create a new {@link HttpRequestBuilder} with GET method and path.
```

Comment:
Could you rephrase the Javadoc a little? This method doesn't really create a new builder anymore 😄